pkg/eigenState/operatorPISplits/operatorPISplits.go

```diff
@@ -0,0 +1,277 @@
+	return nil
+}
+
+func (ops *OperatorPISplitModel) CleanupProcessedStateForBlock(blockNumber uint64) error {
+	delete(ops.stateAccumulator, blockNumber)
+	return nil
+}
+
+func (ops *OperatorPISplitModel) HandleStateChange(log *storage.TransactionLog) (interface{}, error) {
+	stateChanges, sortedBlockNumbers := ops.GetStateTransitions()
+
+	for _, blockNumber := range sortedBlockNumbers {
+		if log.BlockNumber >= blockNumber {
+			ops.logger.Sugar().Debugw("Handling state change", zap.Uint64("blockNumber", log.BlockNumber))
+
+			change, err := stateChanges[blockNumber](log)
+			if err != nil {
+				return nil, err
+			}
+			if change == nil {
+				return nil, nil
+			}
+			return change, nil
+		}
+	}
+	return nil, nil
+}
+
+// prepareState prepares the state for commit by adding the new state to the existing state.
+func (ops *OperatorPISplitModel) prepareState(blockNumber uint64) ([]*OperatorPISplit, error) {
+	accumulatedState, ok := ops.stateAccumulator[blockNumber]
+	if !ok {
+		err := xerrors.Errorf("No accumulated state found for block %d", blockNumber)
+		ops.logger.Sugar().Errorw(err.Error(), zap.Error(err), zap.Uint64("blockNumber", blockNumber))
+		return nil, err
+	}
+
+	recordsToInsert := make([]*OperatorPISplit, 0)
+	for _, split := range accumulatedState {
+		recordsToInsert = append(recordsToInsert, split)
+	}
+	return recordsToInsert, nil
+}
+
+// CommitFinalState commits the final state for the given block number.
+func (ops *OperatorPISplitModel) CommitFinalState(blockNumber uint64) error {
+	recordsToInsert, err := ops.prepareState(blockNumber)
+	if err != nil {
+		return err
+	}
+
+	if len(recordsToInsert) > 0 {
+		for _, record := range recordsToInsert {
+			res := ops.DB.Model(&OperatorPISplit{}).Clauses(clause.Returning{}).Create(&record)
+			if res.Error != nil {
+				ops.logger.Sugar().Errorw("Failed to insert records", zap.Error(res.Error))
+				return res.Error
+			}
+		}
+	}
+	return nil
+}
+
+// GenerateStateRoot generates the state root for the given block number using the results of the state changes.
+func (ops *OperatorPISplitModel) GenerateStateRoot(blockNumber uint64) (types.StateRoot, error) {
+	inserts, err := ops.prepareState(blockNumber)
+	if err != nil {
+		return "", err
+	}
+
+	inputs := ops.sortValuesForMerkleTree(inserts)
+
+	if len(inputs) == 0 {
+		return "", nil
+	}
+
+	fullTree, err := ops.MerkleizeState(blockNumber, inputs)
+	if err != nil {
+		ops.logger.Sugar().Errorw("Failed to create merkle tree",
+			zap.Error(err),
+			zap.Uint64("blockNumber", blockNumber),
+			zap.Any("inputs", inputs),
+		)
+		return "", err
+	}
+	return types.StateRoot(utils.ConvertBytesToString(fullTree.Root())), nil
+}
+
+func (ops *OperatorPISplitModel) sortValuesForMerkleTree(splits []*OperatorPISplit) []*base.MerkleTreeInput {
+	inputs := make([]*base.MerkleTreeInput, 0)
+	for _, split := range splits {
+		slotID := base.NewSlotID(split.TransactionHash, split.LogIndex)
+		value := "added"
```

Comment:
`split.OldOperatorAVSSplitBips` rather than "added"

Reply:
What is the purpose of the value field here?

Reply:
Done in [997e1b3](https://github.com/Layr-Labs/sidecar/pull/112/commits/997e1b36c45bcf628de0e2e26696f535d57fff24). I've updated it with what I think is a better merkle leaf value.